### ISSUE TITLE
Don't run pylint by default even if installed

### ISSUE THREAD

**Issue by ptrcnull** [martinpitt/python-dbusmock#155]
tests for 0.28.4 fail because `mockobject.py` was not linted correctly:

```
..............................................s.................************* Module dbusmock.mockobject
dbusmock/mockobject.py:355:0: R0022: Useless option value for 'disable', 'no-self-use' was moved to an optional extension, see https://pylint.pycqa.org/en/latest/whatsnew/2/2.14/summary.html#removed-checkers. (useless-option-value)
dbusmock/mockobject.py:455:17: C3001: Lambda expression assigned to a variable. Define a function using the "def" keyword instead. (unnecessary-lambda-assignment)

------------------------------------------------------------------
Your code has been rated at 9.96/10

Es...............................ssssssssss.sssssssssssssssssss....ssssss.......ssss..sssssssssssss.....
======================================================================
ERROR: test_pylint (tests.test_code.StaticCodeTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_code.py", line 39, in test_pylint
    subprocess.check_call([sys.executable, '-m', 'pylint'] + glob.glob('dbusmock/*.py'))
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'pylint', 'dbusmock/__main__.py', 'dbusmock/__init__.py', 'dbusmock/mockobject.py', 'dbusmock/testcase.py']' returned non-zero exit status 24.

----------------------------------------------------------------------
```

i'm not really sure why pylint has to be launched from tests, but if you include lints in your testsuite, please check them before releasing a new version :p

**Comment by martinpitt** [comment]
I do check them -- they run in CI. But only on Fedora stable, not on Rawhide any more. Recently pylint changed behaviour wrt. that `no-self-use` thing, and there just isn't a way to make the code happy with *both* 2.13 and 2.14.

Where do you see this failure exactly? It's not in any of the distro builds (pylint should not run anywhere there, it changes too often) or upstream CI.

**Comment by ptrcnull** [comment]
thanks for the quick response!

> there just isn't a way to make the code happy with both 2.13 and 2.14

ohh, okay, that makes way more sense now

> Where do you see this failure exactly?

running `python3 setup.py test` on Alpine Linux edge (pylint 2.14.4)

> pylint should not run anywhere there

well, it's being picked up by the test runner and while it's not a huge problem while packaging for distributions (unless pytest is pulled in by something), it would be cool if lints were not included in the test suite by default

**Comment by martinpitt** [comment]
Fixed in PR #156
